store/pages.go

```diff
@@ -17,46 +27,72 @@ type basicPageType interface {
 	StoreIndexToRecord(idx int, record *Record)
 }
 
+// We use this same struct for storing timestamp value pairs and for
+// storing just timestamps so that we can use a given page for either data
+// structure without needing to use the "unsafe" package.
+//
+// If we are storing only timestamps, we always store two timestamps in each
+// struct: One in the TimeStamp field and one in the Value field.
+// The last struct is the exception and may have either 1 or 2 timestamps.
+// if the Value field of the last struct is nil, it means the last struct
+// has only one timestamp.
+type tsValueType struct {
+	TimeStamp float64
+	Value     interface{}
+}
+
 // a single page of timestamps
-type tsPageType []float64
+type tsPageType []tsValueType
```

Comment:
I think will take more memory for pages with metrics (which are the majority) because a variable of interface type is the size of 2 uintptr.